src/language/lexer.js

```diff
@@ -28,6 +28,10 @@ export class Lexer {
    */
   token: Token;
 
+  /**
+   * The list of comments token encountered so far
+   */
+  commentsList: Array<CommentNode>;
```

Comment:
It makes lexer stateful and also doesn't fit the goals of the lexer.
Lexer should produce tokens and parser should produce AST nodes.

Reply:
Okay, should I then iterate over the tokens produced by lexer which will include the comment tokens too?
It would add additional O(N) complexity then...

Reply:
No just move comment skipping functionality into the parser.